app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt

```diff
@@ -283,3 +305,128 @@ private fun FavoriteParkingCard(
         })
   }
 }
+
+@Composable
+private fun UserReviewsSection(
+    reviewViewModel: ReviewViewModel,
+    userViewModel: UserViewModel,
+    parkingViewModel: ParkingViewModel
+) {
+  val userState = userViewModel.currentUser.collectAsState().value
+
+  if (userState != null) {
+    LaunchedEffect(userState.public.userId) {
+      reviewViewModel.getReviewsByOwnerId(userState.public.userId)
+    }
+  }
+  val userReviews = reviewViewModel.userReviews.collectAsState().value
+
+  Text(
+      text = stringResource(R.string.view_profile_screen_my_reviews_title),
+      style = MaterialTheme.typography.titleLarge,
+      modifier = Modifier.testTag("UserReviewsTitle"))
+
+  Spacer(modifier = Modifier.height(12.dp))
+
+  if (userReviews.isEmpty()) {
+    Text(
+        text = stringResource(R.string.view_profile_screen_no_reviews_message),
+        style = MaterialTheme.typography.bodyMedium,
+        modifier = Modifier.testTag("NoReviewsMessage"))
+  } else {
+    LazyRow(
+        modifier = Modifier.fillMaxWidth().testTag("UserReviewsList"),
+        horizontalArrangement = Arrangement.spacedBy(8.dp)) {
+          itemsIndexed(userReviews) { _, review ->
+            review?.let { ReviewCard(review = it, parkingViewModel = parkingViewModel) }
+          }
+        }
+  }
+}
+
+@Composable
+private fun ReviewCard(review: Review, parkingViewModel: ParkingViewModel) {
```

Comment:
Sorry for being picky but maybe a few comments here would be useful for us to grasp what the function does.

Reply:
added